base/linalg/special.jl

```diff
@@ -124,3 +124,21 @@ end
 
 A_mul_Bc!(A::AbstractTriangular, B::Union{QRCompactWYQ,QRPackedQ}) = A_mul_Bc!(full!(A), B)
 A_mul_Bc(A::AbstractTriangular, B::Union{QRCompactWYQ,QRPackedQ}) = A_mul_Bc(copy!(similar(parent(A)), A), B)
+
+# fill[stored]! methods
+fillstored!(A::Diagonal, x) = (fill!(A.diag, x); A)
+fillstored!(A::Bidiagonal, x) = (fill!(A.dv, x); fill!(A.ev, x); A)
+fillstored!(A::Tridiagonal, x) = (fill!(A.dl, x); fill!(A.d, x); fill!(A.du, x); A)
+fillstored!(A::SymTridiagonal, x) = (fill!(A.dv, x); fill!(A.ev, x); A)
+
+_small_enough(A::Bidiagonal) = size(A, 1) <= 1
+_small_enough(A::Tridiagonal) = size(A, 1) <= 2
+_small_enough(A::SymTridiagonal) = size(A, 1) <= 2
+
+# TODO: Add Diagonal to this method when 0.7 deprecations are removed
+function fill!(A::Union{Bidiagonal,Tridiagonal,SymTridiagonal}, x)
+    xT = convert(eltype(A), x)
+    (xT == zero(eltype(A)) || _small_enough(A)) && return fillstored!(A, xT)
+    throw(ArgumentError("array A of type $(typeof(A)) and size $(size(A)) can
```

Comment:
Perhaps nix the `A`? The user will not be aware of the array's name in the method signature :).